### PR TITLE
YUPTOO-8 Added prometheus-client for metrics collection

### PR DESCRIPTION
**Pull request by patilsuraj767**

### CODE REVIEW

**Review by r14chandra — CHANGES_REQUESTED** [review]

**Comment by r14chandra** [comment]
[L133-l148](https://github.com/RedHatInsights/yuptoo/blob/df6fdc374362118703834c528806a5b636301f19/yuptoo/processor/report_processor.py#L133-L148). In my opinion we can have metrics for `total_hosts_failure` and `total_hosts_processed` to keep the track of hosts. Any suggestions?

File: yuptoo/lib/metrics.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+kafka_failures = Counter(
+    "yuptoo_kafka_failures",
+    "Total number of kafka failures while processing messages",
+    ["account_number"]
+)
+
+HOSTS_COUNTER = Counter(
+    'yupana_hosts_count',
+    'Total number of hosts uploaded',
+    ['source']
```
Comment (kgaikwad) [inline yuptoo/lib/metrics.py:29]:
```suggestion
HOSTS_COUNTER = Counter(
    'yupana_hosts_count',
    'Total number of hosts uploaded',
    ['source', 'org_id']
```
I think we can also keep org_id / account_number along with `source`

File: main.py [hunk 2]
```diff
@@ -42,6 +47,9 @@ def set_extra_log_data(request_obj):
         consumer.commit()
         LOG.error(f"Unable to decode kafka message: {msg.value()}")
     except QPCKafkaMsgException as message_error:
+        kafka_failures.labels(
+            account_number=msg['account']
+        ).inc()
         LOG.error(f"Error processing records.  Message: {msg}, Error: {message_error}")
         consumer.commit()
     except Exception as err:
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply main.py:55]: Added

File: yuptoo/processor/utils.py [hunk 3]
```diff
@@ -49,12 +49,14 @@ def download_report(consumed_message):
         download_response = requests.get(report_url)
 
         LOG.info(f"Successfully downloaded TAR from {report_url}")
+        archive_downloaded_success.inc()
         return download_response.content
 
     except FailDownloadException as fail_err:
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply yuptoo/processor/utils.py]: I have removed the specific `FailDownloadException` and catching it in general `Exception`

File: yuptoo/lib/metrics.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+archive_downloaded_success = Counter(
+    "yuptoo_archive_downloaded_success",
+    "Total number of archives downloaded successfully"
+)
+
+archive_failed_to_download = Counter(
+    "yuptoo_archive_failed_to_download",
+    "Total number of archives that failed to download",
+    ["account_number"]
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply yuptoo/lib/metrics.py]: Removed `account number` and changed it with `org_id`

File: yuptoo/lib/metrics.py (hunk 1 above)
Comment (patilsuraj767) [reply yuptoo/lib/metrics.py]: Done

File: yuptoo/processor/report_processor.py [hunk 5]
```diff
@@ -178,11 +179,17 @@ def extract_report_slices(report_tar, request_obj):
             'Tar does not contain valid JSON metadata & report files'
         )
     except FailExtractException as qpc_err:
+        report_extract_failures.labels(
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply yuptoo/processor/report_processor.py]: I have changed the name of metric

**Review by r14chandra — APPROVED** [review]
Looks good! 👍🏻

**Review by kgaikwad — APPROVED** [review]
LGTM now :+1:

**Pull request merged**